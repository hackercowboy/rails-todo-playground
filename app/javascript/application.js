/* eslint-disable import/prefer-default-export */
import React from 'react';
import ReactDOM from 'react-dom';
import Todos from './components/Todos';

const e = React.createElement;

export function App() {
  return e('main', null, [
    e('h1', { key: 'header' }, 'Rails Todo Playground'),
    e('div', { key: 'container', className: 'container' }, e(Todos)),
  ]);
}

ReactDOM.render(
  e(App),
  document.getElementById('root'),
);
