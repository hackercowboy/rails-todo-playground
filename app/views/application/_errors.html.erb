<ul class="error-list">
  <% @errors.each do |error| %>
    <li class="error-list-item"><%= error %></li>
  <% end %>  
</ul>