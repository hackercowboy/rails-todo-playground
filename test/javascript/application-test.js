import React from 'react';
import { shallow } from 'enzyme';

import { App } from '../../app/javascript/application';

jest.mock('react-dom');

const e = React.createElement;

describe('application', () => {
  it('should work', () => {
    const wrapper = shallow(e(App));
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
