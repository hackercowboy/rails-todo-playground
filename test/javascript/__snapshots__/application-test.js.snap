// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`application should work 1`] = `
"<main>
  <h1>
    Rails Todo Playground
  </h1>
  <div className=\\"container\\">
    <Todos />
  </div>
</main>"
`;
